framework/test/driver/ServiceRegistryTest.cpp

```diff
@@ -80,6 +83,46 @@ void TestMultipleServiceRegistrations(BundleContext context)
   US_TEST_CONDITION_REQUIRED(!ref, "Testing for invalid service reference")
 }
 
+#ifdef US_ENABLE_THREADING_SUPPORT
+// Spawned thread repeatedly calls the IsConvertibleTo method and when the main thread
+// unregisters the service, it used to result in a crash.
+void TestUnregisterFix(BundleContext context)
+{
+  struct TestServiceA : public ITestServiceA
+  {
+  };
+
+  std::atomic_bool done = ATOMIC_VAR_INIT(false);
+  auto invokeIsConvertibleTo = [](ServiceReference<ITestServiceA> ref, std::atomic_bool& done) {
+    while (!done)
+    {
+      (void)ref.IsConvertibleTo("IBooService");
+    }
+  };
+
+  ServiceRegistration<ITestServiceA> registration = context.RegisterService<ITestServiceA>(std::make_shared<TestServiceA>());
+  ServiceReference<ITestServiceA> reference = context.GetServiceReference<ITestServiceA>();
+  std::thread thread(invokeIsConvertibleTo, reference, std::ref(done));
+  std::this_thread::sleep_for(std::chrono::milliseconds(10));
+  registration.Unregister();
+  std::this_thread::sleep_for(std::chrono::milliseconds(10));
+  done = true;
+  thread.join();
+}
+#else
+void TestUnregisterFix(BundleContext context)
```

Comment:
What I meant is that the original bug you are fixing (the `registration->service` nullptr access) is not specific to a multi-threaded test case. We do not need multiple threads to test the correct behavior, but the simple test below is enough. So the multi-threaded version above could be removed completely.

Reply:
Sure.. The reason for two threads test was that it represented a crash we were encountering internally (multi-threaded use case). I'll make the changes.